File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    score = {n : y for n, y in zip(name, yearning)}
    for p in photo:
        memory_score = 0
        for person in p:
            if not score.get(person):
                continue
            memory_score += score[person]
        answer.append(memory_score)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < name.length; i++) {
            map.put(name[i], yearning[i]);
        }
        
        int[] answer = new int[photo.length];
        int idx = 0;
        for (String[] p : photo) {
            int score = 0;
            for (String person : p) {
                if (map.getOrDefault(person, 0) == 0) continue;
                score += map.get(person);
            }
            answer[idx++] = score;
        }
        return answer;
    }
}